### PR TITLE
Fix NPE in UploadService

### LINKED ISSUE #10001: NullPointerException: Attempt to read from field 'org.wordpress.android.fluxc.store.UploadStore org.wordpress.android.u...
Sentry Issue: https://sentry.io/share/issue/08675d53782c40848e2e0882c18bce7d/

```
NullPointerException: Attempt to read from field 'org.wordpress.android.fluxc.store.UploadStore org.wordpress.android.ui.uploads.UploadService.mUploadStore' on a null object reference
    at org.wordpress.android.ui.uploads.UploadService.isPostUploadingOrQueued(UploadService.java:398)
    at org.wordpress.android.ui.posts.PostListUploadStatusTracker.getUploadStatus(PostListUploadStatusTracker.kt:18)
    at org.wordpress.android.ui.posts.PostListMainViewModel$getPostListViewModelConnector$1.invoke(PostListMainViewModel.kt:233)
    at org.wordpress.android.ui.posts.PostListMainViewModel$getPostListViewModelConnector$1.invoke(PostListMainViewModel.kt:60)
    at org.wordpress.android.viewmodel.posts.PostListViewModel.transformPostModelToPostListItemUiState(PostListViewModel.kt:194)
...
(20 additional frame(s) were not displayed)

Attempt to read from field 'org.wordpress.android.fluxc.store.UploadStore org.wordpress.android.ui.uploads.UploadService.mUploadStore' on a null object reference
```

### PR DESCRIPTION
**Pull request by malinajirka**
Fixes #10001 

Fixes NPE in UploadService. We check that `sInstance` field is not null. However, before we use that field `onDestroy` method which clears the field might be have been called by the system. This PR fixes the issue by temporarily storing the `sInstance` into a local field.

To test:
This issue isn't easy to reproduce. I'd suggest just carefully reviewing the code.

Update release notes:

- The changes are too minor.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/uploads/UploadService.java [hunk 1]
```diff
@@ -395,7 +396,7 @@ public static boolean isPostUploadingOrQueued(PostModel post) {
         }
```
[the remark this answers is not in the document]
Comment (malinajirka) [reply WordPress/src/main/java/org/wordpress/android/ui/uploads/UploadService.java]: >I think the race condition stems from this synchronized call within isPostUploadingOrQueued. I'm not sure if, as an alternative to the local variables, we can synchronize here too? (maybe synchronizing on sInstance can reach deadlock?)

I'm not sure I follow, can you please elaborate? Thanks!

My understanding was that we check `instance == null`, but other thread (eg. main thread) might call `onDestroy` before we reach the point where the current thread uses `instance` -> which can result in NPE, since the Main Thread has just cleared the `instance`.

>One question though, will there be any undefined behavior if we access the service instance after its super.onDestroy() has been called?

Good question! Imo it's still a regular class and moreover, we actually use the instance just to access `UploadStore` so I don't think there'll be any undefined behavior.

[the remark this answers is not in the document]
Comment (mkevins) [reply WordPress/src/main/java/org/wordpress/android/ui/uploads/UploadService.java]: > I'm not sure I follow, can you please elaborate? Thanks!

In walking the code path for a blocking call on one thread that would generally allow another thread to "see" a different state, I found that synchronized block. Of course a race condition can occur anyway, as execution order on those separate threads is not guaranteed.. but I was looking specifically for something that could "slow one thread down", or be blocking in some way, making the race condition more likely. I was thinking that may be another approach to consider would be to synchronize, rather than keep a local copy, but my hesitation with that "alternative suggestion" of mine is around the potential for dead-lock.

**TL;DR** At first I had a thought we could synchronize to solve this, but have second thoughts because of the other synchronized block.

Btw, I think your approach is simple and effective. I'm mentioning this synchronized approach mainly for completeness.

>  Imo it's still a regular class and moreover, we actually use the instance just to access UploadStore so I don't think there'll be any undefined behavior.

Ok, I thought that might be the case, thanks for confirming! :+1:
